tests/integration/browser.migration.js

```diff
@@ -631,523 +741,508 @@ describe('migration', function () {
+              return newPouch.allDocs({keys: ['b74e3b45'], include_docs: true});
+            }).then(function (res) {
+              var first = res.rows[0];
+              should.equal(first.value.deleted, true, 'all docs value.deleted');
+              first.value.rev.should.equal('6-441f43a31c89dc68a7cc934ce5779bf8');
+              res.total_rows.should.equal(0);
+              return newPouch.info();
+            }).then(function (info) {
+              info.doc_count.should.equal(0, 'doc_count is 0');
+              return newPouch.changes({include_docs: true, return_docs: true});
+            }).then(function (changes) {
+              changes.results.should.have.length(1);
+              var first = changes.results[0];
+              first.doc._rev.should.equal('6-441f43a31c89dc68a7cc934ce5779bf8');
+              should.equal(first.deleted, true, 'changes metadata.deleted');
+              should.equal(first.doc._deleted, true, 'changes doc._deleted');
+            });
           });
+        }
+      });
+
+      describe('indexeddb', () => {
+        before(function () {
+          if (!usingIndexeddb()) {
+            return this.skip();
+          }
         });
-      }
-    });
 
-    function oldVersionGte(minimumRequired) {
-      const match = scenario.match(/^PouchDB v([.\d]+)$/);
-      if (!match) { return false; }
-      const actual = match[1].split('.');
+        if (versionGte(scenario, '7.2.1')) {
+          it('Should support indexeddb->indexeddb upgrades', () => {
+            const attFor = (id) => new Blob([ 'parentId:' + id ], { type: 'text/plain' });
 
-      const min = minimumRequired.split('.');
+            const oldPouch = new dbs.first.pouch(dbs.first.local, { adapter: 'indexeddb' });
+            let newPouch;
 
-      for (let i=0; i<min.length; ++i) {
-        if (actual[i] > min[i]) { return true; }
-        if (actual[i] < min[i]) { return false; }
-      }
+            return oldPouch
+              .bulkDocs([
+                { _id:'_local/loc-1', some_value:1 },
+                { _id:'norm-1', some_other_value:2 },
+              ])
 
-      return true;
-    }
+              .then(() => oldPouch.get('_local/loc-1'))
+              .then((doc) => oldPouch.putAttachment('_local/loc-1', 'att.txt', doc._rev,  attFor('loc-1'),  'text/plain'))
+
+              .then(()    => oldPouch.putAttachment('_local/loc-2', 'att.txt', undefined, attFor('loc-2'),  'text/plain'))
+
+              .then(()    => oldPouch.get('norm-1'))
+              .then((doc) => oldPouch.putAttachment('norm-1',       'att.txt', doc._rev,  attFor('norm-1'), 'text/plain'))
+
+              .then(()    => oldPouch.putAttachment('norm-2',       'att.txt', undefined, attFor('norm-2'), 'text/plain'))
+
+              .then(() => oldPouch.close())
+
+              .then(() => {
+                newPouch = new dbs.second.pouch(dbs.second.local, { auto_compaction: false, adapter: 'indexeddb' });
+              })
```

Comment:
I can't help but think this sort of setup should be common to all migration tests but this implementation looks quite different from the "idb" one. Ideally as it's an integration test it would be more black box style testing where the "given" and "then" is the same, it's just the parameters for the "when" (adapter, pouch version, etc) that change. Is there a clean way to pull this out so we're testing the same things between adapters?

Reply:
> Is there a clean way to pull this out so we're testing the same things between adapters?

How about https://github.com/pouchdb/pouchdb/pull/8766?